src/tools/blast/components/results/HSPDetailPanel.tsx

```diff
@@ -0,0 +1,383 @@
+loadWebComponent('protvista-manager', ProtvistaManager);
+
+type UniProtkbAccessionsAPI = {
+  results: UniProtkbAPIModel[];
+};
+
+export type HSPDetailPanelProps = {
+  hsp: BlastHsp;
+  hitAccession: string;
+  onClose: () => void;
+  hitLength: number;
+  queryLength: number;
+};
+
+const HSPDetailPanel: FC<HSPDetailPanelProps> = ({
+  hsp,
+  hitAccession,
+  onClose,
+  hitLength,
+  queryLength,
+}) => {
+  const {
+    hsp_align_len,
+    hsp_query_from,
+    hsp_query_to,
+    hsp_qseq,
+    hsp_hit_from,
+    hsp_hit_to,
+    hsp_hseq,
+    hsp_identity,
+  } = hsp;
+  // TODO calculate actual length based on total match and query lengths
+  // const actualLength = hsp_align_len;
+  const actualLength = hitLength;
+  const [highlightPosition, setHighlighPosition] = useState('');
+  const [annotation, setAnnotation] = useState<FeatureType>();
+  const [highlightProperty, setHighlightProperty] = useState<MsaColorScheme>();
+  const [, setActiveTrack] = useState<string>();
+
+  type SegmentRange = {
+    start: number | null;
+    end: number | null;
+  };
+
+  const findSequenceSegments = (seq: string) => {
+    const ranges: number[][] = [];
+    const newRange = () => ({
+      start: null,
+      end: null,
+    });
+
+    let range: SegmentRange = newRange();
+
+    [...seq].forEach((ch, i) => {
+      if (ch !== '-') {
+        if (range.start === null) {
+          range.start = i + 1;
+        }
+      } else if (range.start !== null && range.end === null) {
+        range.end = i;
+        ranges.push([range.start, range.end]);
+        range = newRange();
+      }
+
+      if (i === seq.length - 1 && range.end === null) {
+        range.end = i + 1;
+        ranges.push([range.start as number, range.end as number]);
+        range = newRange();
+      }
+    });
+
+    return ranges;
+  };
+
+  const setQueryTrackData = useCallback(
+    (node): void => {
+      if (node) {
+        const seqSegments = findSequenceSegments(hsp_qseq);
+        const offset = Math.abs(hsp_hit_from - hsp_query_from);
+        const opacity = hsp_identity / 100;
+
+        const blockSegments = seqSegments.map(([start, end]) => ({
+          start: start + offset,
+          end: end + offset,
+          // franklin $colour-sapphire-blue
+          color: '#014371',
+          opacity,
+        }));
+
+        const lineSegments = [
+          {
+            start: offset + 1,
+            end: hsp_query_from + 1,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+          {
+            start: blockSegments[blockSegments.length - 1].end + 1,
+            end: queryLength + offset,
+            shape: 'line',
+            color: '#014371',
+            opacity,
+          },
+        ];
+
+        // eslint-disable-next-line no-param-reassign
+        node.data = [...lineSegments, ...blockSegments];
+      }
+    },
+    [hsp_query_from, hsp_query_to, hsp_identity, hitLength]
```

Comment:
Missing variables in the array of values that the effect depends on - see eslint output to know which ones.

Reply:
Done